### PR TITLE
Component Audit - VMware api e2e provisioning test

### PR DESCRIPTION
**Pull request by shubhamsg199**
Adding VMware e2e api provisioning test.

### CODE REVIEW

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6800
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py --external-logging --include-onprem-provisioning
Test Result : ================ 24 failed, 484 warnings in 2819.35s (0:46:59) =================
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    module_location,
+    pxe_loader,
+    module_vmware_cr,
+    module_vmware_hostgroup,
+    provision_method,
+    vmware,
+    vmwareclient,
+):
+    """Provision a host on vmware compute resource with
+    the help of hostgroup.
```
Comment (shweta83) [inline tests/foreman/api/test_computeresource_vmware.py:43]: Shouldn't this test be Critical instead of High?

**Review by shweta83 — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_computeresource_vmware.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        1. Configure provisioning setup.
+        2. Create VMware CR
+        3. Configure host group setup.
+        4. Provision a host on VMware
+        5. Verify created host on VMware with wrapanapi
+
+    :expectedresults: Host is provisioned succesfully with hostgroup
+
+    :CaseAutomation: Automated
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:57]:
```suggestion
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    host = sat.api.Host(
+        hostgroup=module_vmware_hostgroup,
+        organization=module_sca_manifest_org,
+        location=module_location,
+        name=name,
+        operatingsystem=module_provisioning_rhel_content.os,
+        subnet=module_provisioning_sat.subnet,
+        compute_attributes={
+            "path": "/Datacenters/SatQE-Datacenter/vm/",
+            "cpus": 2,
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:72]: Please prefer single quotes wherever possible

File: tests/foreman/api/test_computeresource_vmware.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        },
+        provision_method=provision_method,
+        build=True,
+    ).create(create_missing=False)
+
+    request.addfinalizer(lambda: sat.provisioning_cleanup(host.name))
+    hostname = f'{name}.{module_provisioning_sat.domain.name}'
+    assert hostname == host.name
+    # check if vm is created on vmware
+    assert vmwareclient.does_vm_exist(hostname) is True
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:104]:
```suggestion
    assert host.name == f'{name}.{module_provisioning_sat.domain.name}'
    # check if vm is created on vmware
    assert vmwareclient.does_vm_exist(host.name) is True
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    # check the build status
+    wait_for(
+        lambda: host.read().build_status_label != 'Pending installation',
+        timeout=1500,
+        delay=10,
+    )
+    host = host.read()
+    assert host.build_status_label == 'Installed'
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:112]:
```suggestion
    assert host.read().build_status_label == 'Installed'
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+from robottelo.config import settings
+
+
+@pytest.mark.e2e
+@pytest.mark.on_premises_provisioning
+@pytest.mark.parametrize('setting_update', ['destroy_vm_on_host_delete=True'], indirect=True)
+@pytest.mark.parametrize('vmware', ['vmware7', 'vmware8'], indirect=True)
+@pytest.mark.parametrize('pxe_loader', ['bios', 'uefi'], indirect=True)
+@pytest.mark.parametrize('provision_method', ['build', 'bootdisk'])
+@pytest.mark.rhel_ver_match('[^6]')
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:27]: Can we split execution of tests with EL8 and EL9 on API and CLI?

Comment (shubhamsg199) [reply tests/foreman/api/test_computeresource_vmware.py:27]: This would be a good idea and would also reduce the provisioning pipeline execution time.

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6916
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py --external-logging --include-onprem-provisioning
Test Result : ================ 8 failed, 1098 warnings in 14661.73s (4:04:21) ================
```

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7132
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py --external-logging --include-onprem-provisioning
Test Result : ================== 27 warnings, 8 errors in 785.32s (0:13:05) ==================
```

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7158
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py --external-logging --include-onprem-provisioning
Test Result : ================ 8 failed, 1102 warnings in 14591.55s (4:03:11) ================
```

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7172
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py --external-logging --include-onprem-provisioning
Test Result : ================ 8 failed, 1102 warnings in 14488.92s (4:01:28) ================
```

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py
provisioning: true
pod_resources_size: extra_extra_large

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py
provisioning: true
pod_resources_size: extra_extra_large

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py
provisioning: true
pod_resources_size: extra_extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7202
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py --external-logging --include-onprem-provisioning
Test Result : ============ 2 failed, 6 passed, 564 warnings in 7830.09s (2:10:30) ============
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+        1. Configure provisioning setup.
+        2. Create VMware CR
+        3. Configure host group setup.
+        4. Provision a host on VMware
+        5. Verify created host on VMware with wrapanapi
+
+    :expectedresults: Host is provisioned succesfully with hostgroup
+
+    :CaseImportance: Critical
```
Comment (shweta83) [inline tests/foreman/api/test_computeresource_vmware.py:57]: Please add one more docstring `:Verifies: SAT-23417`

**Review by shweta83 — APPROVED** [review]

File: tests/foreman/api/test_computeresource_vmware.py (hunk 7 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_computeresource_vmware.py:57]: Updated

**Comment by shubhamsg199** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py -k test_computeresource_vmware.py
provisioning: true
env:
     BROKER_SSH_BACKEND: hussh
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7363
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_computeresource_vmware.py --external-logging --include-onprem-provisioning
Test Result : ============ 2 failed, 6 passed, 545 warnings in 7239.29s (2:00:39) ============
```

**Comment by shubhamsg199** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py -k test_computeresource_vmware.py
provisioning: true
env:
     BROKER_SSH_BACKEND: hussh
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7372
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_computeresource_vmware.py --external-logging --include-onprem-provisioning
Test Result : ============ 2 failed, 6 passed, 550 warnings in 7285.65s (2:01:25) ============
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+            'cpus': 2,
+            'memory_mb': 6000,
+            'cluster': f'{settings.vmware.cluster}',
+            'start': '1',
+            'guest_id': 'rhel8_64Guest',
+            'volumes_attributes': {
+                '0': {
+                    'size_gb': 10,
+                    'thin': '1',
+                    'storage_pod': f'{settings.vmware.datastore_cluster}',
```
Comment (lhellebr) [inline tests/foreman/api/test_computeresource_vmware.py:83]: Does the cluster we use contain multiple pods?

**Review by lhellebr — APPROVED** [review]
ACK with one question regarding validity of testing. The reproducer in BZ says the cluster must have multiple pods.

File: tests/foreman/api/test_computeresource_vmware.py (hunk 8 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_computeresource_vmware.py:83]: Yes, we have multiple datastores forming the cluster

Comment (lhellebr) [reply tests/foreman/api/test_computeresource_vmware.py:83]: Awesome

**Comment by shubhamsg199** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_vmware.py -k test_computeresource_vmware.py
provisioning: true
env:
     BROKER_SSH_BACKEND: hussh
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7444
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_computeresource_vmware.py -k test_computeresource_vmware.py --external-logging --include-onprem-provisioning
Test Result : ============ 2 failed, 6 passed, 537 warnings in 7617.95s (2:06:57) ============
```

File: tests/foreman/api/test_computeresource_vmware.py [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        4. Provision a host on VMware
+        5. Verify created host on VMware with wrapanapi
+
+    :expectedresults: Host is provisioned succesfully with hostgroup
+
+    :CaseImportance: Critical
+
+    :Verifies: SAT-23417
+
+    :BZ: 2186114
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_computeresource_vmware.py:61]: Add :customerscenario: true as well

Comment (shubhamsg199) [reply tests/foreman/api/test_computeresource_vmware.py:63]: Updated

**Comment by shubhamsg199** [comment]
PRT is expected to fail because of known provisioning issue, I've run locally where it is successful. I suggest to merge this as eventually this will be fixed with the bug

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**